src/test/java/seedu/pennywise/logic/parser/AddCommandParserTest.java

```diff
@@ -121,12 +123,27 @@ public void parse_allFieldsPresent_success() {
 
     @Test
     public void parse_optionalFieldsMissing_success() {
-        // zero tag
+        // zero tags
         Entry expectedExpenditure = new ExpenditureBuilder(MOVIE).build();
         assertParseSuccess(parser, TYPE_EXPENDITURE + DESC_MOVIE + AMT_MOVIE + DATE_MOVIE + TAG_MOVIE,
                 new AddCommand(expectedExpenditure, expenditureType));
 
-        // to add income
+        Entry expectedIncome = new seedu.pennywise.testutil.IncomeBuilder(ALLOWANCE).build();
+        assertParseSuccess(parser, TYPE_INCOME + DESC_ALLOWANCE
+                + AMT_ALLOWANCE + DATE_ALLOWANCE + TAG_ALLOWANCE, new AddCommand(expectedIncome, incomeType));
+
+        // no date specified
+        Date today = new Date();
+        String currentDate = " " + PREFIX_DATE + java.time.LocalDate.now()
+                .format(java.time.format.DateTimeFormatter.ofPattern("dd-MM-yyyy"));
+
```

Comment:
Created a default date constructor within the date class, that creates the day based on today's date.